transform/flatten_mangler.go

```diff
@@ -11,6 +12,8 @@ import (
 // DialsTagName is the name of the dials tag.
 const DialsTagName = "dials"
 
+var textMValue = reflect.TypeOf((*encoding.TextUnmarshaler)(nil)).Elem()
```

Comment:
Hmm, `textMValue` is a slightly misleading name. (since it's a type)
Let's call it `textMType` or `textMReflectType`, so we don't confuse ourselves.

Reply:
Good call, also exposed it since it's used in the `flags` package as well

Reply:
The new name is much better.

I'm not sure I like exporting this here. It's not really supposed to be part of the public interface. It's an internal detail. the `TextUnmarshaler` type is also used in `ptrify`, and a couple other subpackages in dials. So far the approach we've been using has been to avoid exposing these `reflect.Type`s by c&ping them into the packages that need them.

If we want to export symbols and share them, I think I'd rather have a separate subpackage expressly for the purpose of tracking these types on which everything else can depend. (probably with unexported package-level variables and functions to get the values so they're actually immutable)

Reply:
Ok unexposed them for now and we can have a separate PR if we want to share them